### PR TITLE
feat: implement email verification in front end

### PR DESCRIPTION
**Pull request by wreality**
This PR adds the front-end functionality for email verification.

**Make sure mailhog is up and running to catch the emails**

See #87 for acceptance criteria

Walkthrough:
- [x] Register a new user
- [x] Click "Resend email" link in the banner (there should now be two emails in mailhog)
- [x] Click the verify user link in the resulting email.

### CODE REVIEW

File: client/src/components/atoms/EmailVerificationSendButton.vue [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        );
+        if (!errorMessages.length) {
+          errorMessages.push(this.$t("failures.UNKNOWN_ERROR"));
+        }
+        this.$q.notify({
+          color: "negative",
+          message: this.$t("account.email_verify.send_failure_notify", {
+            errors: errorMessages.join(", ")
+          }),
+          icon: "bomb"
```
Comment (wreality) [inline client/src/components/atoms/EmailVerificationSendButton.vue:73]:
```suggestion
          icon: "error"
```

File: client/src/components/atoms/EmailVerificationSendButton.jest.spec.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+const notify = jest.fn();
+describe("EmailVerificationSendButton", () => {
+  const wrapper = mountQuasar(EmailVerificationSendButton, {
+    quasar: {
+      components: { QIcon, QBtn, QSpinnerHourglass }
+    },
+    mount: {
+      type: "full",
+      mocks: {
+        $t: (token, _) => token,
```
Comment (wreality) [inline client/src/components/atoms/EmailVerificationSendButton.jest.spec.js:15]:
```suggestion
        $t: token => token,
```

**Review by gmeben — COMMENTED** [review]
I encounter the following error after creating a new user when clicking the Resend Email button: 
```
Email verification could not be sent: An unknown error has occurred
```

File: client/src/i18n/en-us/index.js [hunk 3]
```diff
@@ -60,6 +61,25 @@ export default {
...
   },
   account: {
+    failures: {
+      VERIFY_TOKEN_INVALID:
+        "This verification link is not valid for your account.",
+      VERIFY_EMAIL_VERIFIED: "Email address has already been verified.",
+      VERIFY_TOKEN_EXPIRED: "This verification link has expired."
+    },
+    email_verify: {
+      send_failure_notify: "Email verification could not be sent: {errors}",
```
[the remark this answers is not in the document]
Comment (wreality) [reply client/src/i18n/en-us/index.js:71]: The convention I've adopted it to use uppercase for tokens that are matched against a token from the backend, and lowercase for tokens that are only in the frontend (and so can be refactored without backend updates)

File: client/src/components/atoms/EmailVerificationSendButton.vue [hunk 4]
```diff
@@ -0,0 +1,80 @@
+<template>
+  <q-btn
+    :loading="status == 'loading'"
+    :color="noColor ? null : status == 'success' ? 'positive' : null"
```
[the remark this answers is not in the document]
Comment (wreality) [reply client/src/components/atoms/EmailVerificationSendButton.vue]: I refactored to make it easier to grep, I think.  Let me know what you think.

**Review by gmeben — APPROVED** [review]

**Pull request merged**